### DISCUSSION TITLE
Frequency comparison failure due to units

### DISCUSSION THREAD

**Discussion by fffrune** [scikit-rf/scikit-rf#788] [Q&A] [answered]
I am trying to create a Circuit and all of my networks have the same frequency range, but one is defined in units of Hz, and the other in GHz, causing an error:

```python
tline freqs: 1000000000.0-40000000000.0 Hz, 391 pts
stubs freqs: 1.0-40.0 GHz, 391 pts
Traceback (most recent call last):
  File "subnetwork_growth.py", line 234, in <module>
    SLL_circuit = skrf.Circuit(SLL_connections)
  File "C:\data\school\grad\research\env\lib\site-packages\skrf\circuit.py", line 187, in __init__
    raise AttributeError('All Networks must have same frequencies')
AttributeError: All Networks must have same frequencies

```
I think skrf should be able to compare across frequency units easily?

**Comment by fffrune** [comment, marked as answer]
Nevermind, something else is wrong here.  You can ignore this (I don't know how to delete the issue?).  I'm exporting S-parameters from different applications, the frequency start and stop are the same, but there is something subtly wrong here.

```python
tline freqs: 1.0-40.0 GHz, 391 pts
stubs freqs: 1.0-40.0 GHz, 391 pts
Traceback (most recent call last):
  File "subnetwork_growth.py", line 234, in <module>
    SLL_circuit = skrf.Circuit(SLL_connections)
  File "C:\data\school\grad\research\env\lib\site-packages\skrf\circuit.py", line 187, in __init__
    raise AttributeError('All Networks must have same frequencies')
AttributeError: All Networks must have same frequencies
```

When I compare `tline.f == stubs.f` I get...
```
[ True  True  True  True  True  True  True  True  True  True  True  True
  True  True  True  True  True  True  True  True  True  True  True  True
  True  True  True  True  True  True  True False  True False  True  True
 False  True False  True  True False  True False  True  True False  True
 False  True  True False  True False False False False False False False
  True False False False False False False False False False  True False
 False False False False False False False False  True False False False
 False False False False False False  True  True  True  True  True  True
  True  True  True  True  True  True  True  True  True  True  True  True
  True  True  True  True  True  True  True  True  True  True  True  True
  True  True  True  True  True  True  True  True  True  True  True  True
  True  True  True  True  True  True  True  True  True  True  True  True
  True  True  True  True  True  True  True  True  True  True  True  True
  True  True  True  True  True  True  True  True  True  True  True  True
  True  True  True  True  True  True  True  True  True  True  True  True
  True  True  True  True  True  True  True  True  True  True  True  True
  True  True  True  True  True  True  True  True  True  True  True  True
  True  True  True  True  True  True  True  True  True  True False False
 False False False False  True False False False False False False False
 False False  True False False False False False False False False False
  True False False False False False False False False False  True False
 False False False False False False False False  True False False False
 False False False False False False  True False False False False False
 False False False False  True False False False False False False False
 False False  True False False False False False False False False False
  True False False False False False False False False False  True False
 False False False False False False False False  True False False False
 False False False False False False  True False False False False False
 False False False False  True False False False False False False False
 False False  True False False False False False False False False False
  True False False False False False False False False False  True False
 False False False False False False False False  True False False False
 False False False False False False  True]
```

So it's not an issue with skrf!

**Comment by jhillairet** [comment]
Note that in such a case, scikit-rf can be a great help. You can interpolate a Network using the frequency array used in another Network. See for example [Network.interpolate](https://scikit-rf.readthedocs.io/en/latest/api/generated/skrf.network.Network.interpolate.html) and example [here](https://scikit-rf.readthedocs.io/en/latest/tutorials/Networks.html#Interpolation-and-Concatenation).

I'm moving the issue as a solved question in the discussion part.

**Reply by fffrune** [reply]
I followed the example you linked, e.g. networkA.interpolate(freqs), where networkA already has frequencies very similar to freqs (as above, both  1.0-40.0 GHz, 391 pts), and got the following error:

```python
Traceback (most recent call last):
  File "stub_loaded_lines.py", line 467, in <module>
    SLL_circuit = link_stubs(fred, tee)
  File "stub_loaded_lines.py", line 181, in link_stubs
    tee_network.interpolate(freqs)
  File "C:\data\school\grad\research\env\lib\site-packages\skrf\network.py", line 2720, in interpolate
    x_new = f_interp(f, x, axis=0, **kwargs)(f_new)
  File "C:\data\school\grad\research\env\lib\site-packages\scipy\interpolate\polyint.py", line 78, in __call__
    y = self._evaluate(x)
  File "C:\data\school\grad\research\env\lib\site-packages\scipy\interpolate\interpolate.py", line 684, in _evaluate
    below_bounds, above_bounds = self._check_bounds(x_new)
  File "C:\data\school\grad\research\env\lib\site-packages\scipy\interpolate\interpolate.py", line 716, in _check_bounds
    raise ValueError("A value in x_new is above the interpolation "
ValueError: A value in x_new is above the interpolation range.
```

It looks a lot like the problem this person reported (with an explanation but no solution):
https://stackoverflow.com/questions/45429831/valueerror-a-value-in-x-new-is-above-the-interpolation-range-what-other-re

**Reply by fffrune** [reply]
Ah it looks like it was a different detail again.  I am creating a new network from imported S-parameters, like so:
`extended_network = skrf.Network(frequency=network.f, s=extended_sparams, name=network.name + 'extended')`
where `network.frequency == 1.0-40.0 GHz, 391 pts`
and it turns out that `extended_network.frequency` ended up having the GHz unit multiplier applied twice, i.e. `extended_network.frequency == 1000000000.0-40000000000.0 GHz, 391 pts`
So I should use `network.frequency` instead of `network.f` when creating my `extended_network.`

**Reply by mhuser** [reply]
Hello,

Have you tried adding `fill_value="extrapolate` in `network.interpolate()` ?

**Reply by mhuser** [reply]
```Python
Ntwk.interpolate([...], kind='cubic', fill_value="extrapolate)
```

**Reply by jhillairet** [reply]
> Ah it looks like it was a different detail again. I am creating a new network from imported S-parameters, like so: `extended_network = skrf.Network(frequency=network.f, s=extended_sparams, name=network.name + 'extended')` where `network.frequency == 1.0-40.0 GHz, 391 pts` and it turns out that `extended_network.frequency` ended up having the GHz unit multiplier applied twice, i.e. `extended_network.frequency == 1000000000.0-40000000000.0 GHz, 391 pts` So I should use `network.frequency` instead of `network.f` when creating my `extended_network.`

Yes, indeed, there are currently some inconsistencies in the frequency units depending on how they are passed, ie. as an array or as the Frequency object (recommanded way). The plan is to solve these inconsistencies for the version 1.0.
